skfda/preprocessing/_function_transformers.py

```diff
@@ -0,0 +1,250 @@
+
+import numpy as np
+from sklearn.preprocessing import FunctionTransformer
+
+from skfda.representation.grid import FDataBasis, FDataGrid
+
+from ..exploratory.stats._functional_transformers import (
+    local_averages,
+    number_up_crossings,
+    occupation_measure,
+)
+from ..representation._typing import NDArrayFloat, NDArrayInt
+
+
+class LocalAveragesTransformer(FunctionTransformer):
+    """
+    Transformer that works as an adapter for the local_averages function.
+
+    Args:
+        n_intervals: number of intervals we want to consider.
+
+    Example:
+        We import the Berkeley Growth Study dataset.
+        We will use only the first 30 samples to make the
+        example easy.
+        >>> from skfda.datasets import fetch_growth
+        >>> dataset = fetch_growth(return_X_y=True)[0]
+        >>> X = dataset[:30]
+
+        Then we decide how many intervals we want to consider (in our case 2)
+        and call the function with the dataset.
+        >>> import numpy as np
+        >>> from skfda.preprocessing import LocalAveragesTransformer
+        >>> local_averages = LocalAveragesTransformer(2)
+        >>> np.around(local_averages.fit_transform(X), decimals=2)
+        array([[[ 116.94],
+                [ 111.86],
+                [ 107.29],
+                [ 111.35],
+                [ 104.39],
+                [ 109.43],
+                [ 109.16],
+                [ 112.91],
+                [ 109.19],
+                [ 117.95],
+                [ 112.14],
+                [ 114.3 ],
+                [ 111.48],
+                [ 114.85],
+                [ 116.25],
+                [ 114.6 ],
+                [ 111.02],
+                [ 113.57],
+                [ 108.88],
+                [ 109.6 ],
+                [ 109.7 ],
+                [ 108.54],
+                [ 109.18],
+                [ 106.92],
+                [ 109.44],
+                [ 109.84],
+                [ 115.32],
+                [ 108.16],
+                [ 119.29],
+                [ 110.62]],
+               [[ 177.26],
+                [ 157.62],
+                [ 154.97],
+                [ 163.83],
+                [ 156.66],
+                [ 157.67],
+                [ 155.31],
+                [ 169.02],
+                [ 154.18],
+                [ 174.43],
+                [ 161.33],
+                [ 170.14],
+                [ 164.1 ],
+                [ 170.1 ],
+                [ 166.65],
+                [ 168.72],
+                [ 166.85],
+                [ 167.22],
+                [ 159.4 ],
+                [ 162.76],
+                [ 155.7 ],
+                [ 158.01],
+                [ 160.1 ],
+                [ 155.95],
+                [ 157.95],
+                [ 163.53],
+                [ 162.29],
+                [ 153.1 ],
+                [ 178.48],
+                [ 161.75]]])
+
+    """
+
+    def __init__(self, n_intervals: int):
+        self.n_intervals = n_intervals
```

Comment:
Instead of adding parameters and overloading the `transform` function, just call the parent `__init__` with the appropriate arguments.

The alternative is not to inherit from `FunctionTransformer` and just make this a fully-fledged transformer, but that looks like an overkill to me.